dao/src/main/java/uk/gov/moj/sdt/dao/GenericDao.java

```diff
@@ -1,539 +1,283 @@
+ * or business interruption). However caused any on any theory of liability, whether in contract,
+ * strict liability, or tort (including negligence or otherwise) arising in any way out of the use of this
+ * software, even if advised of the possibility of such damage.
+ *
+ * $Id: $
+ * $LastChangedRevision: $
+ * $LastChangedDate: $
+ * $LastChangedBy: $ */
+package uk.gov.moj.sdt.dao;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.DataAccessException;
+import org.springframework.data.repository.CrudRepository;
+import uk.gov.moj.sdt.dao.api.IGenericDao;
+import uk.gov.moj.sdt.domain.api.IDomainObject;
+import uk.gov.moj.sdt.utils.mbeans.SdtMetricsMBean;
+
+import java.lang.reflect.Array;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+/**
+ * Implements generic DAO functionality based on {@link IGenericDao} allowing a single DAO class to provide access to
+ * all database entities. The DAO an object tree of domain objects. Business logic implemented in service layer acts
+ * upon domain objects returned by JPA. These are translated into JAXB objects before being returned to the SOAP
+ * layer.
+ *
+ * <p>
+ * Subclasses of this class may build upon this to provide additional functionality encapsulating custom queries and
+ * custom SQLs.
+ * </p>
+ *
+ * @author Robin Compston
+ */
+// @Transactional (propagation = Propagation.MANDATORY)
+// @Transactional
+public class GenericDao<DT extends IDomainObject> implements IGenericDao {
+    /**
+     * Logger object.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(GenericDao.class);
+
+    private final CrudRepository<DT, Long> crudRepository;
+
+    private EntityManager entityManager;
+
+    public GenericDao(final CrudRepository<DT, Long> crudRepository) {
+        super();
+        this.crudRepository = crudRepository;
+    }
+
+    /**
+     * Default constructor for {@link GenericDao}.
+     */
+    public GenericDao(final CrudRepository<DT, Long> crudRepository,
+                      EntityManager entityManager) {
+        super();
+        this.crudRepository = crudRepository;
+        this.entityManager = entityManager;
+    }
+
+    public EntityManager getEntityManager() {
+        return entityManager;
+    }
+
+    @Override
+    public <DomainType extends IDomainObject> DomainType fetch(final Class<DomainType> domainType, final long id)
+            throws DataAccessException {
+        // Record start time.
+        final long startTime = new GregorianCalendar().getTimeInMillis();
+
+        LOGGER.debug("fetch(): domainType={}, id=", domainType, id);
+
```

Comment:
The PerformanceLogger code has been removed from this class.  Is this intentional?

Reply:
Yes, we don't need that logging